### PR TITLE
Add status-checker Github Actions

### PR DESCRIPTION
**Pull request by cham11ng**
Add status-checker Github Actions
- Runs lint
- Runs prettier:check
- Runs build
- Runs sonar scanner

### CODE REVIEW

**Review by samirsilwal — APPROVED** [review]

**Review by samirsilwal — APPROVED** [review]

File: status-checker/action.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      shell: bash
+
+    - name: Run build
+      if: ${{ contains(inputs.skip-build, 'true') }}
+      run: |
+        cd ${{ inputs.path }}
+        yarn build
+      shell: bash
+
+    - name: Running security scans (Sonarqube)
```
Comment (mesaugat) [inline status-checker/action.yml:63]:
```suggestion
    - name: Security Scan (Sonarqube)
```

File: status-checker/action.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        yarn lint
+      shell: bash
+
+    - name: Run Prettier Check
+      run: |
+        cd ${{ inputs.path }}
+        yarn prettier:check
+      shell: bash
+
+    - name: Run build
```
Comment (mesaugat) [inline status-checker/action.yml:56]:
```suggestion
    - name: Build
```

File: status-checker/action.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        NODE_AUTH_TOKEN: ${{ inputs.token }}
+      shell: bash
+
+    - name: Run lint
+      run: |
+        cd ${{ inputs.path }}
+        yarn lint
+      shell: bash
+
+    - name: Run Prettier Check
```
Comment (mesaugat) [inline status-checker/action.yml:50]:
```suggestion
    - name: Prettier Check
```

File: status-checker/action.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  steps:
+    - name: Install Dependencies
+      run: |
+        cd ${{ inputs.path }}
+        yarn
+      env:
+        NODE_AUTH_TOKEN: ${{ inputs.token }}
+      shell: bash
+
+    - name: Run lint
```
Comment (mesaugat) [inline status-checker/action.yml:44]:
```suggestion
    - name: Lint
```

File: status-checker/README.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# Status Checker
+
+Setup vpn with openvpn
```
Comment (mesaugat) [inline status-checker/README.md:3]: Not sure I understand. VPN connection should be done by the action that is using status-checker?

File: status-checker/README.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# Status Checker
```
Comment (mesaugat) [inline status-checker/README.md:1]: Let's make the name `status-check` only.

File: status-checker/README.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+Setup vpn with openvpn
+
+### Inputs Required
```
Comment (mesaugat) [inline status-checker/README.md:5]: Need a usage example of this action in the README file.

File: status-checker/README.md (hunk 6 above)
Comment (cham11ng) [reply status-checker/README.md:1]: I'll fix this.

File: status-checker/action.yml [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+    - name: Run build
+      if: ${{ contains(inputs.skip-build, 'true') }}
+      run: |
+        cd ${{ inputs.path }}
+        yarn build
+      shell: bash
+
+    - name: Running security scans (Sonarqube)
+      if: ${{ contains(inputs.skip-security-scanning, 'true') }}
```
Comment (mesaugat) [inline status-checker/action.yml:64]: Mistake?

```suggestion
      if: ${{ contains(inputs.skip-security-scanning, 'false') }}
```

File: status-checker/action.yml [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+      shell: bash
+
+    - name: Run Prettier Check
+      run: |
+        cd ${{ inputs.path }}
+        yarn prettier:check
+      shell: bash
+
+    - name: Run build
+      if: ${{ contains(inputs.skip-build, 'true') }}
```
Comment (mesaugat) [inline status-checker/action.yml:57]: Are we missing a not (!) here?

File: status-check/action.yml [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+        yarn lint
+      shell: bash
+
+    - name: Code Format Check
+      run: |
+        cd ${{ inputs.path }}
+        yarn prettier:check
+      shell: bash
+
+    - name: Build
```
Comment (mesaugat) [inline status-check/action.yml:56]: I also suggest you to add "yarn.lock changes action in here".

Comment (cham11ng) [reply status-check/action.yml:56]: WORK detected. 😢

Comment (cham11ng) [reply status-check/action.yml:67]: Added.

**Review by mesaugat — APPROVED** [review]

**Pull request merged**